app/src/main/java/com/apollo/timeflow/ui/compose/MainActivityByCompose.kt

```diff
@@ -0,0 +1,148 @@
+import com.apollo.timeflow.ui.compose.screenAdaptation.Card
+import com.apollo.timeflow.viewmodel.MainViewModel
+import com.apollo.timeflow.viewmodel.MainViewModelProviderFactory
+import kotlinx.coroutines.flow.collectLatest
+
+class MainActivityByCompose : AppCompatActivity() {
+    private lateinit var mainViewModel: MainViewModel
+    private lateinit var timeChangeReceiver: TimeBroadcast
+    private lateinit var dateChangeReceiver: DateBroadcast
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
+        mainViewModel = ViewModelProvider(
+            this, MainViewModelProviderFactory()
+        )[MainViewModel::class.java]
+
+        this.addBroadcast()
+
+        val mContext = applicationContext ?: return
+        val view = ComposeView(mContext)
+
+        view.setContent {
+            ComposeUI()
+        }
+        setContentView(view)
+
+        mainViewModel.updateDate()
+    }
+
+    @Preview
+    @Composable
+    private fun ComposeUI() {
+        Scaffold { padding ->
+            padding.hashCode()
+
+            val isShowTimeFormatState = remember {
+                mutableStateOf<Boolean?>(null)
+            }
+
+            val isShowDateFormatState = remember {
+                mutableStateOf<Boolean?>(null)
+            }
+
+            val currentDateFormat = remember {
+                mutableStateOf<String?>(null)
+            }
+
+            quickStartCoroutineScope {
+                mainViewModel.timeFormatRecordDataStoreFlow.collectLatest {
+                    isShowTimeFormatState.value = it
+                    mainViewModel.editTimeFormat(it)
+                }
+            }
+
+            quickStartCoroutineScope {
+                mainViewModel.isDateShowDataStoreFlow.collectLatest {
+                    isShowDateFormatState.value = it
+                }
+            }
+
+            quickStartCoroutineScope {
+                mainViewModel.currentDate.collectLatest {
+                    currentDateFormat.value = it
+                }
+            }
+
+            Card(
+                deviceTypes = getDeviceType(),
+                leftOnClick = {
+                    mainViewModel.timeFormatRecordUpdate(!(isShowTimeFormatState.value ?: false))
+                },
+                rightOnClick = {
+                    mainViewModel.isDateShow(!(isShowDateFormatState.value ?: false))
+                },
+                isShowTimeFormat = isShowTimeFormatState,
+                isShowDateFormat = isShowDateFormatState,
+                currentDateFormat = currentDateFormat,
+                hourLeftNumber = mainViewModel.hourLeftNumberState,
+                hourRightNumber = mainViewModel.hourRightNumberState,
+                minuteLeftNumber = mainViewModel.minuteLeftNumberState,
+                minuteRightNumber = mainViewModel.minuteRightNumberState,
+            )
+        }
+    }
```

Comment:
I'm not sure it makes sense to pass in multiple layers of parameters, and then think whether there are any optimized controls

Reply:
Activity it can use view model inside.